pkg/engine/policyset.go

```diff
@@ -320,6 +320,34 @@ func (s *policySet) metadata(ctx context.Context) []MetadataResult {
 	return metadata
 }
 
+// QueryOptions contain options for Engine.Query
+type QueryOptions struct {
```

Comment:
It felt overly verbose to define separate option structs for `Engine.Query` and `policySet.Query` since they'd be identical for now. Here I'm reusing the same struct, but it's documented for `Engine.Query`, because `policySet.query` is package private.